bin/plugin/commands/build-plugins.js

```diff
@@ -0,0 +1,219 @@
+				const pluginVersion = customer[ pluginKey ].version;
+				const pluginSlug = customer[ pluginKey ].slug;
+				try {
+					// Copy module files from the folder.
+					const modulePath = path.join( '.', 'modules/' + key );
+					fs.copySync( modulePath, './build', { overwrite: true } );
+
+					// Update file content.
+					updatePluginFiles( {
+						key: pluginKey,
+						slug: pluginSlug,
+						version: pluginVersion,
+					} );
+
+					// Update since version.
+					updateSinceVersion( {
+						version: pluginVersion,
+					} );
+
+					// Update text domain.
+					updateTextDomain( {
+						slug: pluginSlug,
+					} );
+				} catch ( error ) {
+					log(
+						formats.error(
+							`\n${ error }"\n`
+						)
+					);
+				}
+			}
+		} catch ( jsonError ) {
+			log(
+				formats.error(
+					`\nError parsing JSON string: "${ jsonError }"\n\n`
+				)
+			);
+		}
+	} );
+};
+
+/**
+ * Updates the `load.php` file header information.
+ *
+ * @param {[]Settings} settings Plugin settings.
+ */
+async function updatePluginFiles( settings ) {
+	const loadFile = path.join( '.', 'load.php' );
+	const loadFileContent = fs.readFileSync( loadFile, 'utf-8' );
+	const buildLoadFile = path.join( '.', 'build/load.php' );
+	const buildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	const regex = /\/\\*\\*[\s\S]+?(?=\*\/)/mi;
+	const pluginComment = loadFileContent.match( regex )?.[ 0 ];
+	const newPluginComment = buildLoadFileContent.match( regex )?.[ 0 ];
+
+	fs.writeFileSync( buildLoadFile, buildLoadFileContent.replace( newPluginComment, pluginComment ) );
+
+	let updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	// Get module header data.
+	const { name, description } = await getModuleData( buildLoadFileContent );
+	const moduleData = {
+		pluginname: name,
+		plugindescription: description,
+		pluginuri: '/tree/trunk/modules/' + settings.key,
+		pluginversion: settings.version,
+		textdomain: settings.slug,
+	};
+
+	// Map of module header => object property.
+	const headers = {
+		'Plugin Name': 'pluginname',
+		'Plugin URI': 'pluginuri',
+		Description: 'plugindescription',
+		Version: 'pluginversion',
+		'Text Domain': 'textdomain',
+	};
+
+	const headerRegex = new RegExp( `(?<header>${ Object.keys( headers ).join( '|' ) }): (?<value>(.*?)+)`, 'gmi' );
+	const matches = [ ...updatedBuildLoadFileContent.matchAll( headerRegex ) ];
+
+	matches.forEach( ( match ) => {
+		const { header, value } = match.groups;
+		const prop = headers[ header ];
+		let moduleDataValue = moduleData[ prop ];
+
+		if ( value && prop && moduleDataValue ) {
+			if ( prop === 'pluginuri' ) {
+				moduleDataValue = value + moduleDataValue;
+			}
+			fs.writeFileSync( buildLoadFile, updatedBuildLoadFileContent.replace( new RegExp( value ), moduleDataValue ) );
+			updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+		}
+	} );
+}
+
+/**
+ * Updates the `readme.txt` file with the given changelog.
```

Comment:
@mukeshpanchal27 is this function description correct? `getModuleData` does not appear to update anything, instead it returns data.

Reply:
It also does not perform any changelog related tasks from what I am observing

Reply:
Thanks @10upsimon for review. Addressed in https://github.com/WordPress/performance/pull/662/commits/b19e57cd4f65a36829ddfa24cd1dad3f6771d60b